### PR TITLE
Added u05-sa-softmax

### PR DESCRIPTION
**Pull request by MatthewWalstra**

### CODE REVIEW

**Review by kcarnold — APPROVED** [review]
Improve... compared to what?

**Comment by MatthewWalstra** [comment]
I didn't know the best way to say what are things that softmax does to prepare the neural net outputs for negative log likelihood and why are they important (categorical data only)

**Comment by kcarnold** [comment]
It seems like you're trying to ask several different questions at the same time here. Maybe split it apart. e.g., what might go wrong if we just used exp(logits) and forgot to divide by the sum?
